canonical/maas-ui
# feat(kvm): update resource tables to fetch machines

## Done

- Update the KVM resource tables to fetch the VMs using the new API.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Connect your UI to Bolla.
- Go to KVM -> karura-polong-20211104 -> in the hosts table click 'polong'.
- Turn "View by NUMA node" on.
- In the numa card click "Total VMs: 1" and you should see the table of the VMs in the NUMA node.

## Fixes

Fixes: canonical/app-tribe#1324.

## Screenshots

![Screen Recording 2022-09-07 at 10 36 45 am](https://user-images.githubusercontent.com/361637/188770674-9291fe25-2541-46f4-a801-b7f4126f7d24.gif)
